### PR TITLE
feat: add profile details edit functionality

### LINKED ISSUE #45: feat: add profile details edit functionality
Should be done after auth + backend is done

### PR DESCRIPTION
**Pull request by AyanavaKarmakar**
- [x] update user name
- [x] update user email
- [x] update both user name and email

### CODE REVIEW

**Comment by AyanavaKarmakar** [comment]
Working Demo: https://app.flonnect.com/view/video/mccreeper2k/Flonnect_2023-01-15_cb1851b7-d4c9-407f-8301-553b6f295ad9

File: src/components/Profile/EditProfile.tsx [hunk 1]
```diff
@@ -111,6 +141,7 @@ export const EditProfile = () => {
 
               <div className="mt-4 flex justify-end">
                 <Dialog.Close
+                  onClick={() => handleSave()}
```
Comment (Nishith-Savla) [inline src/components/Profile/EditProfile.tsx]:
```suggestion
                  onClick={handleSave}
```

File: src/server/api/routers/userProfile.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        await ctx.prisma.user.update({
+          where: {
+            id: input.userId,
+          },
+          data: {
+            name: input.newName,
+          },
+        });
+      } catch (error) {
+        console.log(error);
```
Comment (Nishith-Savla) [inline src/server/api/routers/userProfile.ts]:
```suggestion
        console.error(error);
```

File: src/server/api/routers/userProfile.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        await ctx.prisma.user.update({
+          where: {
+            id: input.userId,
+          },
+          data: {
+            email: input.newEmail,
+          },
+        });
+      } catch (error) {
+        console.log(error);
```
Comment (Nishith-Savla) [inline src/server/api/routers/userProfile.ts]:
```suggestion
        console.error(error);
```

File: src/components/Profile/EditProfile.tsx (hunk 1 above)
Comment (AyanavaKarmakar) [reply src/components/Profile/EditProfile.tsx]: @Nishith-Savla 

image

**Review by ssarkar551 — APPROVED** [review]

File: src/components/Profile/EditProfile.tsx (hunk 1 above)
Comment (Nishith-Savla) [reply src/components/Profile/EditProfile.tsx]: Source?

Comment (AyanavaKarmakar) [reply src/components/Profile/EditProfile.tsx]: chatgpt. on second thought, maybe this isn't our use case here, so we can just shorten

Comment (Nishith-Savla) [reply src/components/Profile/EditProfile.tsx]: Ohh okay

**Review by Nishith-Savla — APPROVED** [review]

**Review by Nishith-Savla — APPROVED** [review]

**Pull request merged**